### DISCUSSION TITLE
Cookies always return None in Swagger Docs

### DISCUSSION THREAD

**Discussion by amitness** [fastapi/fastapi#7240] [Questions] [answered]
### First check

* [X] I added a very descriptive title to this issue.
* [X] I used the GitHub search to find a similar issue and didn't find it.
* [X] I searched the FastAPI documentation, with the integrated search.
* [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [X] I already read and followed all the tutorial in the docs and didn't find an answer.
* [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [X] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import FastAPI

app = FastAPI()


@app.get('/cookies')
def cookie(name=Cookie(None)):
    return {'name': name}
```

### Description

* Run the above code
* Goto swagger to test the cookies. 
* The name is returned as null instead of what was input.

![Screenshot from 2020-08-16 14-06-41](https://user-images.githubusercontent.com/8587189/90330110-15c76b80-dfca-11ea-9c87-323adb042b85.png)

### Environment

* OS: [e.g. Linux / Windows / macOS]: Linux
* FastAPI Version [e.g. 0.3.0]: 0.60.1

To know the FastAPI version use:

```bash
python -c "import fastapi; print(fastapi.__version__)"
```

* Python version:

To know the Python version use:

```bash
python --version
```

**Comment by SirTelemak** [comment, marked as answer]
Cookie authentication currently does not work in Swagger UI 
Here you can see why: [browser security restrictions](https://developer.mozilla.org/en-US/docs/Glossary/Forbidden_header_name) 
TL:DR it prevent web pages from modifying cookie programmatically.
Check [this](https://github.com/swagger-api/swagger-js/issues/1163) for more information about this limitation

**Reply by YuriiMotov** [reply]
To test the endpoints that require Cookie parameters, you can create the `set_cookie` endpoint that accepts `cookie_name` and `cookie_value` parameters and set them.
So, first use that new endpoint to set the value of cookie parameter, then use your endpoint that requires cookie parameter. The parameter you set on the first step will be passed to your endpoint

**Comment by amitness** [comment]
Thanks @SirTelemak

**Comment by perkfly** [comment]
Duplicate of #880

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue @amitness  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.

**Comment by mostopalove** [comment]
Hello! I see that this [issue](https://github.com/swagger-api/swagger-ui/issues/8683) was fixed in swaggerUI. 
What version of Swagger is using now in FastAPI?
